sstables/sstable_directory.cc

```diff
@@ -166,12 +180,16 @@ sstable_directory::process_sstable_dir(process_flags flags) {
 
     scan_state state;
 
-    directory_lister sstable_dir_lister(_sstable_dir, lister::dir_entry_types::of<directory_entry_type::regular>(), &manifest_json_filter);
+    components_lister sstable_dir_lister(_sstable_dir);
     std::exception_ptr ex;
     try {
-        while (auto de = co_await sstable_dir_lister.get()) {
-            auto comps = sstables::entry_descriptor::make_descriptor(_sstable_dir.native(), de->name);
-            handle_component(state, std::move(comps), _sstable_dir / de->name);
+        while (true) {
+            sstring name = co_await sstable_dir_lister.get();
+            if (name == "") {
+                break;
+            }
```

Comment:
nit: why not keep the previous `while()` with a condition?

Reply:
```
sstables/sstable_directory.cc:186:21: error: no viable conversion from 'seastar::basic_sstring<char, unsigned int, 15, true>' to 'bool'
        while (auto name = co_await sstable_dir_lister.get()) {
                    ^~~~
/home/xemul/src/scylla-2/seastar/include/seastar/core/sstring.hh:219:5: note: candidate function
    operator std::basic_string<char_type>() const {
    ^
/home/xemul/src/scylla-2/seastar/include/seastar/core/sstring.hh:575:5: note: candidate function
    operator std::basic_string_view<char_type>() const noexcept {
    ^
1 error generated.
```
:man_shrugging:

Reply:
sstring has no operator bool
You can try this:
`while (!(auto name = ...).empty())) {`
But this is stretching it a bit.

Reply:
`for (sstring name; name = ...; !name.empty()) {`
:)

Reply:
> `for (sstring name; name = ...; !name.empty()) {` :)

:rofl:

Reply:
off-topic: Why is there sstring at all? Isn't `std::string` good enough? Git archaeology doesn't give an answer, sstring.hh is there since initial commit, just moves around directories

Reply:
No idea, probably just inertia. We don't really have a reason to use sstring since C++XX, don't recall which one but I know we already on it for some time.
The only reason to use sstring is for `bytes` and similar, where `sstring` has uninitialized ctor and doesn't enforce storing closing `\0`.